djstripe/event_handlers.py

```diff
@@ -14,95 +15,195 @@
+        # customers that do not already exist locally.
+        return
 
+    if crud_type.deleted:
+        # Deletions for customers are handled a little bit differently, since
+        # the customer is "purged" but not fully deleted from the system, so
+        # the object is updated first then "deleted".
+        _handle_crud_type_event(
+            Customer, event_data, event_subtype,
+            crud_type=CrudType(updated=True))
 
-@webhooks.handler(['customer'])
-def customer_webhook_handler(event, event_data, event_type, event_subtype):
+    _handle_crud_type_event(
+        Customer, event_data, event_subtype, crud_type=crud_type)
 
-    customer = event.customer
-    if customer:
-        if event_subtype in STRIPE_CRUD_EVENTS:
-            versioned_customer_data = Customer(stripe_id=event_data["object"]["id"]).api_retrieve()
-            Customer.sync_from_stripe_data(versioned_customer_data)
 
-            if event_subtype == "deleted":
-                customer.purge()
-#         elif event_subtype.startswith("discount."):
-#             pass  # TODO
-        elif event_subtype.startswith("source."):
-            source_type = event_data["object"]["object"]
+@webhooks.handler("customer.source")
+def customer_source_webhook_handler(
+        event, event_data, event_type, event_subtype):
+    """ Handles updates for customer source objects. """
+    source_type = event_data["object"]["object"]
 
-            # TODO: other sources
-            if source_type == "card":
-                versioned_card_data = Card(stripe_id=event_data["object"]["id"], customer=customer).api_retrieve()
-                Card.sync_from_stripe_data(versioned_card_data)
-        elif event_subtype.startswith("subscription."):
-            versioned_subscription_data = Subscription(stripe_id=event_data["object"]["id"], customer=customer).api_retrieve()
-            Subscription.sync_from_stripe_data(versioned_subscription_data)
+    # TODO: other sources
+    if source_type != "card":
+        return
 
+    obj, crud_type = _handle_crud_type_event(
+        Card, event_data, event_subtype, customer=event.customer)
 
-# ---------------------------
-# Transfer model events
-# ---------------------------
-@webhooks.handler(["transfer"])
+
+@webhooks.handler("customer.subscription")
+def customer_subscription_webhook_handler(
+        event, event_data, event_type, event_subtype):
+    """ Handles updates for subscription objects. """
+    _handle_crud_type_event(
+        Subscription, event_data, event_subtype, customer=event.customer)
+
+
+@webhooks.handler("transfer")
 def transfer_webhook_handler(event, event_data, event_type, event_subtype):
-    versioned_transfer_data = Transfer(stripe_id=event_data["object"]["id"]).api_retrieve()
-    Transfer.sync_from_stripe_data(versioned_transfer_data)
+    """ Handles updates for transfer objects. """
+    _handle_crud_type_event(Transfer, event_data, event_subtype)
```

Comment:
for all of these calls to ``_handle_crud_type_event``, please use explicit kwargs